### ISSUE TITLE
Use dB values for faders on Mixer tab

### ISSUE THREAD

**Issue by Lange** [nodecg/nodecg#441]
As suggested by Mark Schwartzkopf.

**Comment by EwanLyon** [comment]
Not entirely sure if this one is possible since the SoundJS volume is linear and dB is a log scale

**Comment by Lange** [comment]
Translation between linear and log are generally possible, but require a knowledge of the specific implementation of the existing linear mechanisms. For instance, it is documented how the 0-1 float values in the protocol of the X32 digital mixer map to actual dB values, and it's a pretty complicated formula.

**Comment by Lange** [comment]
Further reading:
- https://webaudioapi.com/book/Web_Audio_API_Boris_Smus_html/ch03.html
- https://stackoverflow.com/questions/22604500/web-audio-api-working-with-decibels

The first link mentions dBFS, which I believe is the scale we'd want to use here. Moving a volume slider all the way to the right would read as `0 dBFS`, and moving it all the way to the left would read as `-∞ dBFS`.

I'll ask Mark (the person who suggested this change, and a very good audio engineer) to maybe make a GitHub account and pop in here to offer his thoughts and guidance.

**Comment by TestRunnerSRL** [comment]
So this is somewhat hairy since the % volume slider often isn't even linear. Like in the X32 as mentioned, the scale is a piece-wise function, and I think that kinda of scaling is actually somewhat common. Windows has a feature to show the volume sliders in dB instead of %, which honestly is a lot nicer. Many devices have 100% quite a bit above true 0 dB, so being able to see the actual line level is great. When working with any kind of sound mixer, you pretty much always want to work with dB instead of % because it not only means everything is using more consistent metering, but also that it gives better impression of relative audio in general.

If you know the your scaling is truly a linear scale (like 0% to 100%) then calculating the dB is fairly simple. It's basically just `20 * log10(percent_volume)`, which will be between 0dB and -inf. Or above 0dB if it's above 100% volume.

**Comment by Dan-Shields** [comment]
I actually started working on this issue a while ago, then stumbled pretty hard getting the dB<->linear translation. I have the UI done for dB though.

One thing though @TestRunnerSRL, I thought audio was `20 * log10`? You use 20 for power, 10 for voltage, and I assumed audio was thought of as power.

**Comment by TestRunnerSRL** [comment]
Yeah for audio, it's ~6dB for to half/double the signal and for voltage it's ~3dB to half/double. For whatever reason audio is double the scale, hence the `20 * log10`.

**Comment by Dan-Shields** [comment]
I now realise that's exactly what you wrote, apologies, I need to sleep 😩

**Comment by Lange** [comment]
This would be another great feature to add in the upcoming v2.0 release.
